gammapy/cube/new.py

```diff
@@ -235,25 +236,34 @@ def make_map_hadron_acceptance(pointing, livetime, bkg, ref_geom, offset_max):
     background : `~gammapy.maps.WcsNDMap`
         Background predicted counts sky cube in reco energy
     """
-    # Compute offsets of all pixels
-    offset = make_separation_map(ref_geom, pointing).quantity
-
-    # Retrieve energies from WcsNDMap
-    # Note this would require a log_center from the geometry
-    energy = ref_geom.axes[0].center * ref_geom.axes[0].unit
 
     # Compute the expected background
     # TODO: properly transform FOV to sky coordinates
     # For now we assume the background is radially symmetric
 
     # TODO: add a uniform API to the two background classes
+    energy_axis = ref_geom.axes[0]
+    # Compute offsets of all pixels
+    offset = make_separation_map(ref_geom, pointing).quantity
+
     if isinstance(bkg, Background3D):
-        data = bkg.data.evaluate(detx=offset, dety='0 deg', energy=energy)
+        map_coord = ref_geom.get_coord()
+        # Compute offsets of all pixels
+        detx = offset
```

Comment:
This seems incorrect according to the new `Background3D.evaluate`, no?
`offset`here is a 2D array. Shouldn't it be  a 3D array?
In that case you should rather do:
`offset = map_coord.skycoord.separation(pointing)`